skynet/dgpu/compute.py

```diff
@@ -81,153 +69,140 @@ def prepare_params_for_diffuse(
+        we need to raise an exception here if we need to cancel work
+        '''
+        step = args[0]
+        # compat with callback_on_step_end
+        if not isinstance(step, int):
+            step = args[1]
+
+        maybe_update_tui(lambda tui: tui.set_progress(step, done=total_steps))
+
+        should_raise = False
+        if should_cancel:
+            should_raise = trio.from_thread.run(should_cancel, request_id)
+
+        if should_raise:
+            logging.warning(f'CANCELLING work at step {step}')
+            raise DGPUInferenceCancelled('network cancel')
+
+        return {}
+
+    maybe_update_tui(lambda tui: tui.set_status(f'Request #{request_id}'))
+
+    inference_step_wakeup(0)
+
+    output_type = 'png'
+    if 'output_type' in params:
+        output_type = params['output_type']
+
+    output = None
+    output_hash = None
+    try:
+        name = params['model']
+
+        match method:
+            case 'diffuse' | 'txt2img' | 'img2img' | 'inpaint':
+                arguments = prepare_params_for_diffuse(
+                    params, method, inputs)
+                prompt, guidance, step, seed, upscaler, extra_params = arguments
+
+                if 'flux' in name.lower():
+                    extra_params['callback_on_step_end'] = inference_step_wakeup
+
+                else:
+                    extra_params['callback'] = inference_step_wakeup
+                    extra_params['callback_steps'] = 1
+
+                output = model(
+                    prompt,
+                    guidance_scale=guidance,
+                    num_inference_steps=step,
+                    generator=seed,
+                    **extra_params
+                ).images[0]
+
+                output_binary = b''
+                match output_type:
+                    case 'png':
+                        if upscaler == 'x4':
+                            input_img = output.convert('RGB')
+                            up_img, _ = init_upscaler().enhance(
+                                convert_from_image_to_cv2(input_img), outscale=4)
+
+                            output = convert_from_cv2_to_image(up_img)
+
+                        output_binary = convert_from_img_to_bytes(output)
+
+                    case _:
+                        raise DGPUComputeError(f'Unsupported output type: {output_type}')
+
+                output_hash = sha256(output_binary).hexdigest()
+
+            case 'upscale':
+                input_img = inputs[0].convert('RGB')
+                up_img, _ = model.enhance(
+                    convert_from_image_to_cv2(input_img), outscale=4)
+
+                output = convert_from_cv2_to_image(up_img)
+
+                output_binary = convert_from_img_to_bytes(output)
+                output_hash = sha256(output_binary).hexdigest()
+
+            case _:
+                raise DGPUComputeError('Unsupported compute method')
+
+    except BaseException as err:
```

Comment:
not sure if it's relevant but depending on how the above raises is there a diff between an internal error and a set of excs expected from certain sections?

again might be something to consider for tracking our own internal regressions longer term.

Reply:
removed this whole try-except on `structify`